auth0/src/main/java/com/auth0/android/provider/OAuthManager.java

```diff
@@ -154,13 +156,15 @@ public void onSuccess(@NonNull Credentials codeCredentials) {
 
     //Helper Methods
 
-    private void assertNoError(String errorValue) throws AuthenticationException {
+    private void assertNoError(String errorValue, String errorDescription) throws AuthenticationException {
         if (errorValue == null) {
             return;
         }
         Log.e(TAG, "Error, access denied. Check that the required Permissions are granted and that the Application has this Connection configured in Auth0 Dashboard.");
         if (ERROR_VALUE_ACCESS_DENIED.equalsIgnoreCase(errorValue)) {
             throw new AuthenticationException(ERROR_VALUE_ACCESS_DENIED, "Permissions were not granted. Try again.");
+        } else if (ERROR_VALUE_UNAUTHORIZED.equalsIgnoreCase(errorValue)) {
+            throw new AuthenticationException(ERROR_VALUE_UNAUTHORIZED, errorDescription);
```

Comment:
can `errorDescription` be null/empty? in the case of a rule error, this is something we don't control, right? it's user defined.
maybe we can add a generic message "Unknown error" in case it's null, just to avoid a null description?